src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1185,12 +1186,22 @@ protected boolean isCategoryEnabled(@NonNull SCMHeadCategory category) {
      */
     @Override
     public void afterSave() {
-        SCMSourceOwner owner = getOwner();
+        SCMSourceOwner owner = getOwnerNotSynchronized();
         if (owner != null) {
-            GitHubWebHook.get().registerHookFor(owner);
+            getHook().registerHookFor(owner);
         }
     }
 
+    @VisibleForTesting
+    SCMSourceOwner getOwnerNotSynchronized() {
```

Comment:
Just for testing purposes: allows mocking GitHubSCMSource and return some specific directly.